ipatests/test_integration/test_sssd.py

```diff
@@ -108,3 +113,45 @@ def test_auth_cache_enabled_when_configured(self, user):
             assert self.is_auth_cached(self.users[user])
             time.sleep(10)
             assert not self.is_auth_cached(self.users[user])
+
+    @contextmanager
+    def filter_user_setup(self, user):
+        sssd_conf_backup = tasks.FileBackup(self.master, paths.SSSD_CONF)
+        filter_user = {'filter_users': '{0}'.format(self.users[user]['name'])}
+        try:
+            if user == 'fakeuser':
+                provider = 'nss'
+            else:
+                provider = 'ipa'
+            tasks.modify_sssd_conf(self.master, self.master.domain.name,
+                                   filter_user, provider=provider)
+            tasks.clear_sssd_cache(self.master)
+            yield
+        finally:
+            sssd_conf_backup.restore()
+
+    @pytest.mark.xfail(
+        osinfo.id == 'fedora' and osinfo.version_number <= (30,),
+        reason='https://pagure.io/SSSD/sssd/issue/3978')
+    @pytest.mark.parametrize('user', ['ad', 'fakeuser'])
+    def test_filter_users_start_sssd_offline(self, user):
+        """Negative cache should use values from 'filter_users' config option.
+
+        Test for https://bugzilla.redhat.com/show_bug.cgi?id=1685472
+        https://bugzilla.redhat.com/show_bug.cgi?id=1724088
+
+        When there are users in filter_users with domain then no look up
+        should be in data provider.
+        1: domain section for AD user
+        2: for other domain added user in nss section.
+        """
+        with self.filter_user_setup(user=user):
+            log_file = '{0}/sssd_nss.log'.format(paths.VAR_LOG_SSSD_DIR)
+            logsize = tasks.get_logsize(self.master, log_file)
+            self.master.run_command(
+                ['getent', 'passwd', self.users[user]['name']],
+                raiseonerr=False, ok_returncode=[0, 2])
```

Comment:
1. It is a mistake to use both `raiseonerr` and `ok_returncode`. With `raiseonerr=False` the function will not raise exception with any return code.
2. We expect that user is in negative cache, so there is only one valid return code -- 2